### PR TITLE
Replaces Gulp, Browserify and BrowserSync with Webpack

### PR DESCRIPTION
**Pull request by aendrew**
This is working to some extent — running Webpack generates a bundle and a html file. 

## Still ToDo:

- [ ] Get environment change merged into upstream [nunjucks-webpack-plugin](https://github.com/itgalaxy/nunjucks-webpack-plugin) [#4](https://github.com/itgalaxy/nunjucks-webpack-plugin/pull/4)
- [x] Get CSS to generate
- [x] Add support for revving
- [x] Add support for image minification
- [x] Add Bower support
- [x] Enable local development
- [x] Ensure prod performance is as it should be

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ jobs:
   build:
     working_directory: ~/project
     docker:
-      - image: circleci/node:latest
+      - image: node:8
```
Comment (callumlocke) [inline .circleci/config.yml]: Why switch back to this? Isn't CircleCI's own image better?

File: webpack.config.babel.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      environment: nunjucksEnv,
+    }),
+    new CopyWebpackPlugin([
+      { from: 'client/components/core/top.css', to: 'top.css' },
+      { from: 'client/images/*.+(jpg|jpeg|svg|png|gif)', to: 'images/', flatten: true },
+    ], {
+      copyUnmodified: true,
+    }),
+    env.production ? new ImageminWebpackPlugin({ test: /\.(jpe?g|png|gif|svg)$/i }) : undefined,
+    function revReplace() {
```
Comment (callumlocke) [inline webpack.config.babel.js]: Nothing seems to be getting revved. When I do `npm run build` it makes `dist/bundle.js`, so it looks like revving isn't working.

File: views/includes/html-head.html [hunk 3]
```diff
@@ -8,9 +8,9 @@
 <link rel="preconnect" href="https://cdn.polyfill.io">
 
 {# Stylesheets #}
-<link inline rel="stylesheet" href="components/core/top.css">
+<link inline rel="stylesheet" href="top.css">
 <link rel="stylesheet" href="https://www.ft.com/__origami/service/build/v2/bundles/css?modules=o-normalise@^1,o-fonts@^2,o-grid@^4.2.1,o-typography@^4.3.1,o-header@^6.5.1,o-card@^2.1.8,o-footer@^5.0.3{{',o-comments@^3.2.0' if flags.comments}}{{ ',o-share@^3' if flags.shareButtons }}{{ ',o-ads@^6.1.5' if flags.ads}}">
-<link rel="stylesheet" href="styles.css">
+<link rel="stylesheet" href="bundle.css">
```
Comment (callumlocke) [inline views/includes/html-head.html]: This is a 404, in both dev server and prod build.

File: .circleci/config.yml (hunk 1 above)
Comment (aendrew) [inline .circleci/config.yml]: I was getting weird annoying npm bugs with Circle's image, which is why I switched back. I forget what the actual reason was, feel free to switch it back if you wish to investigate.

File: views/includes/html-head.html (hunk 3 above)
Comment (aendrew) [inline views/includes/html-head.html]: This should be fixed now — `env` is a string and not an object, which is why it wasn't revving anything.

**Comment by ajwl** [comment]
Having just got a version of this project built on S3 - can confirm deployment works ✅

File: webpack.config.babel.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  devServer: {
+    hot: true,
+    contentBase: resolve(__dirname, 'client'),
+  },
+  devtool: 'source-map',
+  plugins: [
+    new HotModuleReplacementPlugin(),
+    new ExtractTextPlugin({
+      filename: env === 'production' ? '[name].[contenthash].css' : '[name].css',
+      disable: env !== 'production',
```
Comment (ajwl) [inline webpack.config.babel.js]: Delete this one line `disable: env !== 'production',` - causing a file not to be loaded on local

**Pull request merged**